sdk/tables/data-tables/src/serialization.ts

```diff
@@ -145,13 +145,51 @@ export function deserialize<T extends object = Record<string, any>>(
       if (`${key}@odata.type` in obj) {
         const type = (obj as any)[`${key}@odata.type`];
         typedValue = getTypedObject(value, type, disableTypeConversion);
+      } else if(["number", "string"].includes(typeof value)) {
+        // The service, doesn't return type metadata for number or strings
+        // if automatic type conversion is disabled we'll infer the EDM object
+        typedValue = inferTypedObject(key, value, disableTypeConversion);
       }
+
       deserialized[transformedKey] = typedValue;
     }
   }
   return deserialized;
 }
 
+function inferTypedObject(propertyName: string, value: number | string, disableTypeConversion: boolean) {
+  // Use value as is when typeConversion is enabled
+  if(!disableTypeConversion) {
+    return value;
+  }
```

Comment:
seems like we could avoid calling this method instead of having to have a disable boolean?